src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -490,6 +520,191 @@ end
+
+
+  # Calculate blending factor alpha_blending
+  alpha_blending = volume_integral.indicator
+
+
+  # Calculate volume integral contribution
+  for j in eachnode(dg), i in eachnode(dg)
+    for v in eachvariable(equations)
+      du[v, i, j, element] += ( (1 - alpha_blending) *
+                                (inverse_weights[i] * (fstaggered1[v, i+1, j] - fstaggered1[v, i, j]) +
+                                 inverse_weights[j] * (fstaggered2[v, i, j+1] - fstaggered2[v, i, j])) ) +
+                                ( alpha_blending *
+                                (inverse_weights[i] * (fstar1_L[v, i+1, j] - fstar1_R[v, i, j]) +
+                                 inverse_weights[j] * (fstar2_L[v, i, j+1] - fstar2_R[v, i, j])) )
+
+    end
+  end
+
+  # -----------------------------------------------------------------------------------------------
+  # old version of calculation with fstaggered as Vector of SVectors
+  # (only for one "line", in x or y direction)
+  # @unpack fstaggered_threaded = cache
+
+  # fstaggered = fstaggered_threaded[Threads.threadid()]
+
+  # # x
+  # for j in eachnode(dg)
+  #   # compute local high-order flux
+  #   local_fluxes_1!(fstaggered, u, nonconservative_terms, equations, volume_flux_dg, dg, j, element)
+
+  #   # update volume contribution in locally conservative form
+  #   add_to_node_vars!(du, (1 - alpha_blending) * inverse_weights[1] * fstaggered[1], equations, dg, 1, j, element)
+  #     for i in 2:nnodes(dg)-1
+  #       add_to_node_vars!(du, (1 - alpha_blending) * inverse_weights[i] * (fstaggered[i] - fstaggered[i-1]), equations, dg, i, j, element)
+  #     end
+  #   add_to_node_vars!(du, -(1 - alpha_blending) * inverse_weights[end] * fstaggered[end], equations, dg, nnodes(dg), j, element)
+  # end
+
+  # # y
+  # for i in eachnode(dg)
+  #   # compute local high-order fluxes
+  #   local_fluxes_2!(fstaggered, u, nonconservative_terms, equations, volume_flux_dg, dg, i, element)
+
+  #   # update volume contribution in locally conservative form
+  #   add_to_node_vars!(du, (1 - alpha_blending) * inverse_weights[1] * fstaggered[1], equations, dg, i, 1, element)
+  #     for j in 2:nnodes(dg)-1
+  #       add_to_node_vars!(du, (1 - alpha_blending) * inverse_weights[j] * (fstaggered[j] - fstaggered[j-1]), equations, dg, i, j, element)
+  #     end
+  #   add_to_node_vars!(du, -(1 - alpha_blending) * inverse_weights[end] * fstaggered[end], equations, dg, i, nnodes(dg), element)
+  # end
+
+  return nothing
+end
+
+
+#     calcflux_staggered!(fstaggered1, fstaggered2, u, mesh,
+#                         nonconservative_terms, equations, volume_flux_dg, dg, element, cache)
+#
+# Calculate the staggered grid volume fluxes inside the elements (**without non-conservative terms**).
+#
+# # Arguments
+# - `fstaggered1::AbstractArray{<:Real, 3}`
+# - `fstaggered2::AbstractArray{<:Real, 3}`
+@inline function calcflux_staggered!(fstaggered1, fstaggered2, u::AbstractArray{<:Any,4},
```

Comment:
```suggestion
@inline function calcflux_staggered!(fstaggered1, fstaggered2, u,
```
Shouldn't be necessary